packages/lib/modules/pool/actions/LiquidityActionHelpers.ts

```diff
@@ -103,7 +106,39 @@ export class LiquidityActionHelpers {
   }
 
   public get poolStateWithBalances(): PoolStateWithBalances {
-    return toPoolStateWithBalances(this.pool)
+    return isBoosted(this.pool)
+      ? this.boostedPoolStateWithBalances
+      : toPoolStateWithBalances(this.pool)
+  }
+
+  /* Used by calculateProportionalAmounts for V3 boosted proportional adds */
+  public get boostedPoolStateWithBalances(): PoolStateWithBalances {
+    const underlyingTokensWithBalance: PoolTokenWithBalance[] = this.pool.poolTokens.map(
+      (token, index) =>
+        token.underlyingToken
+          ? {
+              address: token.underlyingToken?.address as Address,
+              decimals: token.underlyingToken?.decimals as number,
+              index,
+              // TODO: balance: token.underlyingToken?.balance * rate as HumanAmount,
+              balance: token.balance as HumanAmount,
```

Comment:
@garethfuller Note that this balance should be the underlying balance (wrappingToken.balance * priceRate) instead of just wrappingToken balance but I'm investigating why that's failing.
Related PR on top of this one: 
https://github.com/balancer/frontend-monorepo/pull/153